File: app/src/main/java/com/example/movicard/Splash.java
package com.example.movicard;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class Splash extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_splash);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        ImageView logo = findViewById(R.id.logo_splash);

        Animation animation = AnimationUtils.loadAnimation(this, R.anim.fade_in_scale);
        logo.startAnimation(animation);

        // Ejecutar el código con un retraso en el hilo principal sin bloquear la interfaz.
        new Handler(Looper.getMainLooper()).postDelayed(() -> {
            Intent intent = new Intent(Splash.this, SingUp.class);
            startActivity(intent);
            finish();
        }, 2000); // 2000 ms = 2 segundos
    }
}

File: MoviCard/app/src/main/java/com/example/movicard/Splash.kt
package com.example.movicard

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.animation.AnimationUtils
import android.widget.ImageView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class Splash : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_splash)

        val logo = findViewById<ImageView>(R.id.logo_splash)

        val animation = AnimationUtils.loadAnimation(this, R.anim.fade_in_scale)
        logo.startAnimation(animation)

        // Ejecutar el código con un retraso en el hilo principal sin bloquear la interfaz.
        Handler(Looper.getMainLooper()).postDelayed({
            val intent = Intent(
                this@Splash,
                Login::class.java
            )
            startActivity(intent)
            finish()
        }, 2000) // 2000 ms = 2 segundos
    }
}